soccer/RobotIntent.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <Constants.hpp>
+#include <Geometry2d/ShapeSet.hpp>
+#include "planning/MotionCommand.hpp"
+#include "planning/RotationCommand.hpp"
+
+typedef std::array<float, Num_Shells> RobotMask;
+
+struct RobotIntent {
+    enum class ShootMode { KICK, CHIP };
+    enum class TriggerMode { STAND_DOWN, IMMEDIATE, ON_BREAK_BEAM };
+    enum class Song { STOP, CONTINUE, FIGHT_SONG };
+
+    std::unique_ptr<Planning::MotionCommand> _motionCommand;
+    std::unique_ptr<Planning::RotationCommand> _rotationCommand;
+
+    /// set of obstacles added by plays
+    Geometry2d::ShapeSet _local_obstacles;
+
+    /// masks for obstacle avoidance
+    RobotMask _opp_avoid_mask;
+    float _avoidBallRadius;  /// radius of ball obstacle
```

Comment:
This is a struct, so you don't need to start things with _. Also, all entries should be consistently snake_case.